src/EditorFeatures/CSharpTest/SimplifyInterpolation/SimplifyInterpolationTests.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.SimplifyInterpolation;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.SimplifyInterpolation
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsSimplifyInterpolation)]
+    public partial class SimplifyInterpolationTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new CSharpSimplifyInterpolationDiagnosticAnalyzer(), new CSharpSimplifyInterpolationCodeFixProvider());
+
+        [Fact]
+        public async Task ToString_with_string_literal_parameter_should_be_shown_as_unnecessary()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(int someValue)
+    {
+        _ = $""prefix {someValue{|Unnecessary:[||].ToString(""|}some format code{|Unnecessary:"")|}} suffix"";
+    }
+}",
+@"class C
+{
+    void M(int someValue)
+    {
+        _ = $""prefix {someValue:some format code} suffix"";
+    }
+}");
+        }
```

Comment:
needs more tests :)

Reply:
I have plenty of tests in mind :D

(see the TODO I added)